servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultServiceDiscoveryRetryStrategy.java

```diff
@@ -261,18 +269,23 @@ void errorSeen() {
     }
 
     private static final class IndefiniteRetryStrategy implements BiIntFunction<Throwable, Completable> {
-        private static final int MAX_RETRIES = 10;
         private final BiIntFunction<Throwable, Completable> delegate;
+        private final int maxRetries;
 
         IndefiniteRetryStrategy(final Executor executor, final Duration initialDelay) {
-            delegate = retryWithExponentialBackoffAndJitter(MAX_RETRIES, __ -> true, initialDelay, executor);
+            this(executor, initialDelay, 10);
+        }
+
+        IndefiniteRetryStrategy(final Executor executor, final Duration initialDelay, final int maxRetries) {
+            delegate = retryWithExponentialBackoffAndJitter(maxRetries, __ -> true, initialDelay, executor);
+            this.maxRetries = maxRetries;
         }
 
         @Override
         public Completable apply(final int count, final Throwable cause) {
             // As we are retrying indefinitely (unless closed), cap the backoff on MAX_RETRIES retries to avoid
             // impractical backoffs
-            return delegate.apply(count % (MAX_RETRIES - 1), cause);
+            return delegate.apply(max(1, count % (maxRetries)), cause);
```

Comment:
With `max` approach it will return `1` two times. Not a big deal. But if you don't like it, alternative could be `(count % MAX_RETRIES) + 1`.
Also, IIRC, the power of 2 value will be a bit more optimal for %. How about 8 or 16 instead of 10? Or just a mask of `0x0F`?